pallets/nft/src/lib.rs

```diff
@@ -290,55 +315,53 @@ pub mod pallet {
             let (did, _) = EnsureDid::<T>::ensure_origin(origin)?;
 
             // 1. ensure funded
+            let instance_id = Self::get_or_create_preferred_nft(&did)?;
 
-            let mut meta = Did::<T>::meta(&did).ok_or(Error::<T>::NotExists)?;
-
-            ensure!(meta.nft.is_none(), Error::<T>::Minted);
+            let mut meta = <NftMetaStore<T>>::get(&instance_id).ok_or(Error::<T>::NotExists)?;
+            ensure!(!meta.minted, Error::<T>::Minted);
 
             let deposit = <T as parami_did::Config>::Currency::free_balance(&meta.pot);
 
+            let init = T::InitialMintingDeposit::get();
             ensure!(
-                deposit >= T::InitialMintingDeposit::get(),
+                deposit >= init,
                 Error::<T>::InsufficientBalance
             );
 
             // 2. create NFT token
+            let tid = instance_id;
 
-            let cid = NextClassId::<T>::try_mutate(|id| -> Result<T::AssetId, DispatchError> {
-                let current_id = *id;
-                *id = id.checked_add(&One::one()).ok_or(Error::<T>::Overflow)?;
-                Ok(current_id)
-            })?;
-
-            let tid = T::AssetId::min_value();
-
-            T::Nft::create_class(&cid, &meta.pot, &meta.pot)?;
-            T::Nft::mint_into(&cid, &tid, &meta.pot)?;
+            T::Nft::create_class(&meta.class_id, &meta.pot, &meta.pot)?;
+            T::Nft::mint_into(&meta.class_id, &tid, &meta.pot)?;
 
             // 3. initial minting
 
             let initial = T::InitialMintingValueBase::get();
 
-            T::Assets::create(cid, meta.pot.clone(), true, One::one())?;
-            T::Assets::set(cid, &meta.pot, name.clone(), symbol.clone(), 18)?;
-            T::Assets::mint_into(cid, &meta.pot, initial.saturating_mul(3u32.into()))?;
+            T::Assets::create(tid, meta.pot.clone(), true, One::one())?;
+            T::Assets::set(tid, &meta.pot, name.clone(), symbol.clone(), 18)?;
+            T::Assets::mint_into(tid, &meta.pot, initial.saturating_mul(3u32.into()))?;
 
             // 4. transfer third of initial minting to swap
 
-            T::Swaps::new(&meta.pot, cid)?;
-            T::Swaps::mint(&meta.pot, cid, deposit, deposit, initial, false)?;
+            T::Swaps::new(&meta.pot, tid)?;
+            T::Swaps::mint(&meta.pot, tid, deposit, deposit, initial, false)?;
 
-            meta.nft = Some(cid);
+            // 5. update local variable
+            meta.minted = true;
 
-            Did::<T>::set_meta(&did, meta);
+            // 6. update storage
+            <NftMetaStore<T>>::mutate(&tid, |maybe| {
+                *maybe = Some(meta);
+            });
```

Comment:
`mutate` equals to `get` then `insert`, it cost 1 read and 1 write. we should use `insert` here due to we already `get` this value.
```suggestion
            <NftMetaStore<T>>::insert(tid, meta);
```